consensus/src/test/java/org/apache/iotdb/consensus/iot/StabilityTest.java

```diff
@@ -108,14 +119,55 @@ public void snapshotTest() throws Exception {
     Assert.assertEquals(1, versionFiles1.length);
 
     consensusImpl.triggerSnapshot(dataRegionId);
-    Thread.sleep(10);
+
     consensusImpl.triggerSnapshot(dataRegionId);
 
     File[] versionFiles2 =
         dataDir.listFiles((dir, name) -> name.startsWith(IoTConsensusServerImpl.SNAPSHOT_DIR_NAME));
+    consensusImpl.deletePeer(dataRegionId);
     Assert.assertNotNull(versionFiles2);
     Assert.assertEquals(1, versionFiles2.length);
 
     Assert.assertNotEquals(versionFiles1[0].getName(), versionFiles2[0].getName());
   }
+
+  public void snapshotUpgradeTest() throws Exception {
+    consensusImpl.createPeer(
+        dataRegionId,
+        Collections.singletonList(new Peer(dataRegionId, 1, new TEndPoint("0.0.0.0", 9000))));
+    consensusImpl.triggerSnapshot(dataRegionId);
+    String oldSnapshotId =
+        String.format(
+            "%s_%s_%d",
+            IoTConsensusServerImpl.SNAPSHOT_DIR_NAME,
+            dataRegionId.getId(),
+            System.currentTimeMillis());
+    File snapshotDir = new File(storageDir, "1_1");
+    File oldSnapshotName = new File(snapshotDir, oldSnapshotId);
+    if (oldSnapshotName.exists()) {
+      org.apache.commons.io.FileUtils.deleteDirectory(oldSnapshotName);
+    }
+    if (!oldSnapshotName.mkdirs()) {
+      throw new ConsensusGroupModifyPeerException(
+          String.format("%s: cannot mkdir for snapshot", dataRegionId));
+    }
+    consensusImpl.triggerSnapshot(dataRegionId);
```

Comment:
Maybe you should check  `Assert.assertFalse(oldSnapshotDir.exists())` after this line